addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRule.java

```diff
@@ -168,76 +174,110 @@ public int getRisk() {
         return Alert.RISK_HIGH;
     }
 
+    @Override
+    public void scan(HttpMessage msg, NameValuePair originalParam) {
```

Comment:
I believe you could keep the class extending  AbstractAppPlugin instead of AbstractAppParamPlugin and use the VariantMultipartFormParameters to obtain the parameter only on multipart requests. This would also solve the problem of storing values in the class properties ( if I understand what you were trying to do)